dune_client/api/table.py

```diff
@@ -0,0 +1,96 @@
+"""
+Table API endpoints enables users to
+create and insert data into Dune.
+"""
+
+from __future__ import annotations
+from typing import List, Dict, Any, IO
+
+from dune_client.api.base import BaseRouter
+from dune_client.models import DuneError
+
+
+class TableAPI(BaseRouter):
+    """
+    Implementation of Table endpoints - Plus subscription only
+    https://docs.dune.com/api-reference/tables/
+    """
+
+    def upload_csv(
+        self,
+        table_name: str,
+        data: str,
+        description: str = "",
+        is_private: bool = False,
+    ) -> bool:
+        """
+        https://docs.dune.com/api-reference/tables/endpoint/upload
+        This endpoint allows you to upload any .csv file into Dune. The only limitations are:
+
+        - File has to be < 200 MB
+        - Column names in the table can't start with a special character or digits.
+        - Private uploads require a Plus subscription.
+
+        Below are the specifics of how to work with the API.
+        """
+        response_json = self._post(
+            route="/table/upload/csv",
+            params={
+                "table_name": table_name,
+                "description": description,
+                "data": data,
+                "is_private": is_private,
+            },
+        )
+        try:
+            return bool(response_json["success"])
+        except KeyError as err:
+            raise DuneError(response_json, "UploadCsvResponse", err) from err
+
+    def create_table(
+        self,
+        namespace: str,
+        table_name: str,
+        schema: List[Dict[str, str]],
+        description: str = "",
+        is_private: bool = False,
+    ) -> Any:
```

Comment:
Why `Any` ?

Reply:
That is what `self._post` returns. 
This was already implemented, but we can change that once we have something returned

Reply:
I am trying to say that `Any` is not a useful return type. What should users expect back when calling the `create_table` or `insert_table` functions ? 

> but we can change that once we have something returned

But we should already have a response returned from the API, right ? 

> That is what self._post returns.

The fact that `self._post` returns `Any`, might have been a bad design / tech debt. If this is part of the public API, which we cannot change, maybe we can consider returning a more useful type here.

If not an easy change, let's keep it like that.

Reply:
I agree that it was probably not the best decision. 
This function is kind of unnecessary, which is used to handle `_post`, `_get`, and `_patch` responses

https://github.com/duneanalytics/dune-client/blob/e3f6c98eab115e4f03d256b18fc8709a524c886f/dune_client/api/base.py#L139-L150 

Should have just returned the `Response` object. 
`response.json()` returns `Any`, so this will take a bit of work. 

And as such, I don't think is worth it.

Reply:
> This was already implemented, but we can change that once we have something returned

I take this back, this was not as simple as I thought it would be.

Reply:
Thanks 🖖

Reply:
Sorry folks, this was my fault ☹️ . But these internal private methods return Any and the idea is to convert them to the expected type inside these pubic facing interfaces. So, for example... we have these Object.from_dict() methods that convert Any responses to the expected types:

https://github.com/cowprotocol/dune-client/blob/b9db1904c4c32768caf3cb5a6dddea45db3f4097/dune_client/api/execution.py#L66-L72

Here would have been the same thing. Since I am currently implementing these API routes in the Node project, I will submit a corresponding PR for this one.

Reply:
Here is the related PR: https://github.com/duneanalytics/dune-client/pull/117